src/utils/useEvents.ts

```diff
@@ -21,35 +21,31 @@ const useEvents = <T extends AllLeafletInstances | Map>(
   let mapInstance = useMapInstance()
 
   if (instance instanceof Map) {
-    // @ts-ignore
     mapInstance = instance
   }
```

Comment:
This code might produce hard-to-track bugs. In the case one wants to use this function for binding events to another typeof instance which is null at the time useEvents is called, the events will get bound to the map, in a next cycle the the other typeof instance might exist and the events will be bound again, this time to the correct typeof instance.

It might be better to add a function:

```
useMapEvents(events) {
  const mapInstance = useMapInstance()
  useEvents(mapInstance, events)
}
```

Reply:
and not automagically bind the events to the map

Reply:
I believe this was introduced by @chrisvanmook in order to pass a map instance in cases we don't have the provider in the hierarchy. Perhaps we can create a separate issue for this since it's not part of this refactor.

Reply:
.. and the useMapEvents function already exists! So.. I'll talk to Chris about it.